### PR TITLE
Rewrite Invitation resolver

### PR DESCRIPTION
**Pull request by yijia-cc**

### CODE REVIEW

File: app/api/gqlv2/resolver/deps.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+type Dependencies struct {
+	userDao dao.User
+	teamDao dao.Team
+}
+
+func NewDependencies(
+	userDao dao.User,
+	teamDao dao.Team,
+) *Dependencies {
```
Comment (naturetime10) [inline app/api/gqlv2/resolver/deps.go]: 1) Why are we returning pointer for dependencies? 
2) Are we exposing dependencies as public API to be consumer in external packages?

File: app/api/gqlv2/resolver/invitation.go [hunk 2]
```diff
@@ -3,53 +3,84 @@ package resolver
...
 )
 
 type Invitation struct {
+	invitation entityv2.Invitation
+	deps       *Dependencies
 }
 
-func (Invitation) ID(ctx context.Context) (graphql.ID, error) {
-	panic("implement me")
+func (i Invitation) ID(ctx context.Context) (graphql.ID, error) {
```
Comment (naturetime10) [inline app/api/gqlv2/resolver/invitation.go]:
```suggestion
func (i Invitation) ID(ctx context.Context) graphql.ID {
```

File: app/api/gqlv2/resolver/invitation.go [hunk 3]
```diff
@@ -3,53 +3,84 @@ package resolver
...
 
+	"github.com/teamyapp/teamy-backend/app/entityv2"
+
 	"github.com/graph-gophers/graphql-go"
-	"github.com/teamyapp/teamy-backend/app/entity"
 )
 
 type Invitation struct {
+	invitation entityv2.Invitation
+	deps       *Dependencies
```
Comment (naturetime10) [inline app/api/gqlv2/resolver/invitation.go]:
```suggestion
	deps       Dependencies
```

File: app/api/gqlv2/resolver/team.go [hunk 4]
```diff
@@ -3,10 +3,14 @@ package resolver
...
 	"context"
 
+	"github.com/teamyapp/teamy-backend/app/dao"
+
 	"github.com/graph-gophers/graphql-go"
 )
 
 type Team struct {
+	team dao.Team
+	deps *Dependencies
```
Comment (naturetime10) [inline app/api/gqlv2/resolver/team.go]:
```suggestion
	deps Dependencies
```

File: app/api/gqlv2/resolver/user.go [hunk 5]
```diff
@@ -3,10 +3,14 @@ package resolver
...
 	"context"
 
+	"github.com/teamyapp/teamy-backend/app/dao"
+
 	"github.com/graph-gophers/graphql-go"
 )
 
 type User struct {
+	user dao.User
+	deps *Dependencies
```
Comment (naturetime10) [inline app/api/gqlv2/resolver/user.go]: Let's separate this into the next PR.

**Review by naturetime10 — CHANGES_REQUESTED** [review]

File: app/api/gqlv2/resolver/deps.go (hunk 1 above)
Comment (yijia-cc) [reply app/api/gqlv2/resolver/deps.go]: 2. NewDependency will be used in wire.go for static DI.

File: app/api/gqlv2/resolver/user.go (hunk 5 above)
Comment (yijia-cc) [reply app/api/gqlv2/resolver/user.go]: Want to have a complete implementation for Invitation resolver.

**Review by naturetime10 — APPROVED** [review]

**Pull request merged**